### ISSUE TITLE
ChangeEntityExperienceEvent doesn't work

### ISSUE THREAD

**Issue by 5igm** [SpongePowered/Sponge#1467]
I want to write a plugin which can edit the exp formula, but the event ChangeEntityExperienceEvent seems cannot be fired when a player gains EXP. How to deal with it?
Code: https://my.pcloud.com/publink/show?code=XZEFXcZRnULoH9L1RzvyRoOPEvdP48t8IGV
There is no error in the log.

**Comment by JBYoshi** [comment]
I have some prototype code ready. It depends on #1565; I will submit the changes after it is merged.

**Comment by phit** [comment]
closing in favor of https://github.com/SpongePowered/SpongeCommon/issues/1569
dont need two issues for this
